### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by Hamayounali**
## Performed Activities
1. Created method current_user
2. Created a new post form
3. Created a comment form
4. Created a Like form

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,69 +1,20 @@
...
-  # GET /comments/new
-  def new
-    @comment = Comment.new
-  end
-
-  # GET /comments/1/edit
-  def edit; end
-
-  # POST /comments or /comments.json
   def create
```
Comment (Lucas-Masaba) [inline app/controllers/comments_controller.rb:2]: - [ ] Please add the new action as mentioned in the requirements. It should look similar to this
```
  def new
    @comment = Comment.new
  end

```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -9,4 +9,32 @@ def show
...
   end
+
+  # GET /posts/new
+  def new
+    @post = Post.new
+  end
+
+  # POST /posts or /posts.json
+  def create
+    @post = Post.new(post_params.merge(comments_counter: 0, likes_counter: 0, author_id: current_user.id))
```
Comment (Lucas-Masaba) [inline app/controllers/posts_controller.rb:20]: - [ ] There is an error when creating a post indicating the id is nil
image

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -9,4 +9,32 @@ def show
...
+    respond_to do |format|
+      if @post.save
+        format.html do
+          redirect_to "/users/#{@post.user.id}", notice: 'Post was successfully created.'
```
Comment (Lucas-Masaba) [inline app/controllers/posts_controller.rb:24]: - [ ] This should be `@post.current_user.id` not `@post.user.id`

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @Hamayounali  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Hamayounali) [reply app/controllers/posts_controller.rb:20]: Hi @Lucas-Masaba, the create post form works fine on my side. please check the below screenshots 
image
after submit 
image

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (Hamayounali) [reply app/controllers/comments_controller.rb:2]: Hi, @Lucas-Masaba , Please see it in the data_model branch, not in development branch
image

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (Hamayounali) [reply app/controllers/posts_controller.rb:24]: The @post.user.id is used to redirect to the posts page according to the route.

**Comment by arslanbisharat** [comment]
Hi @Hamayounali,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**